src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Color DEFAULT_COLOR = Color.WHITE;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int MAX_SIZE = 10;
+    private final ColorSupplier colorSupplier;
+
+    {
+        colorSupplier = new ColorSupplier();
+    }
+
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        Color color = colorSupplier.getRandomColor();
+        int numFigureTypes = 5;
+        int figureType = random.nextInt(numFigureTypes);
+
+        switch (figureType) {
+            case 0:
+                return new Square(random.nextDouble() * MAX_SIZE,color);
+            case 1:
+                return new Rectangle(random.nextDouble() * MAX_SIZE,
+                        random.nextDouble() * MAX_SIZE, color);
+            case 2:
+                return new RightTriangle(random.nextDouble() * MAX_SIZE,
+                        random.nextDouble() * MAX_SIZE, color);
+            case 3:
+                return new Circle(random.nextDouble() * MAX_SIZE, color);
```

Comment:
Better create own methods for creating figures

Reply:
You mean: createRandomSquare(), createRandomRectangle(), createRandomRightTriangle(), createRandomCircle(), and createRandomIsoscelesTrapezoid()? That's not what is written in a README file. "FigureSupplier with the public Figure getRandomFigure() method - for generating figures with random properties."

Reply:
It doesn't contradict Readme, public method will return figures, relying on private methods internally. However, it's up to you whether to extract this logic to a private method. The rule of thumb is readability, which option is easier to grasp for someone who reads it for the first time. I think logic here is rather trivial, and by creating many more private methods we can even lose readability, but that's debatable